File: src/main/scala/com/github/tetrisanalyzer/piece/PieceO.scala
package com.github.tetrisanalyzer.piece

class PieceO extends Piece {
  val number = 7.toByte
  val character = 'O'
  protected val widths = Array(2)
  protected val heights = Array(2)
  protected val shapes = Array(
    new PieceShape(Array(Point(0,0), Point(1,0), Point(0,1), Point(1,1)))
  )
}

File: src/main/java/com/github/tetrisanalyzer/piece/PieceO.java
package nu.tengstrand.tetrisanalyzer.piece;

public class PieceO extends Piece {
    @Override public byte number() { return 7; }
    @Override public char character() { return 'O'; }
    @Override protected int[] widths() { return new int[] { 2 }; }
    @Override protected int[] heights() { return new int[] { 2 }; }
    @Override protected PieceShape[] shapes() {
        return new PieceShape[] {
            new PieceShape(new Point(0,0), new Point(1,0), new Point(0,1), new Point(1,1))
        };
    }
}
